### DISCUSSION TITLE
Unauthenticated access to the serverless APIs

### DISCUSSION THREAD

**Discussion by davidaventimiglia-professional** [supabase/supabase#11356] [Questions] [answered]
Hello!

Evidently at least the `anon` key is need to access the serverless APIs (REST and GraphQL).  Is there any way around this, to allow unauthenticated or anonymous access to those APIs?

I'm thinking in terms of something similar to [unauthenticated access](https://hasura.io/docs/latest/auth/authentication/unauthenticated-access/) in Hasura.  I understand the security concerns involved and why it would be prudent to be cautious about this sort of thing, but I'm wondering if it's even _possible_ in supabase.

**Comment by steve-chavez** [comment, marked as answer]
It is possible with the [db-anon-role ](https://postgrest.org/en/stable/configuration.html#db-anon-role) from postgREST but this is not exposed on the hosted version.

**Reply by davidaventimiglia-professional** [reply]
Thanks!

**Reply by PierreMesure** [reply]
Hi,

I'm trying to setup API routes that do not require any authentication nor API key in Supabase (self-hosted) to download the data (so SELECT should be all open) and this is the only place I could find that even mentions the possibility. I can't believe this isn't a use case for more people. In my case, some of the tables would contain government open data so we don't need API keys nor any form of authentication, we just want to lower the barrier to download the data as much as possible.

I looked into db-anon-role, I couldn't understand in which way it supported this need, and Postgrest doesn't seem to offer the possibility of completely open data either.

Any help would be welcome if I misunderstood the possibilities here! ❤

**Reply by steve-chavez** [reply]
Related to https://github.com/PostgREST/postgrest/discussions/3764

>  and Postgrest doesn't seem to offer the possibility of completely open data either.

PostgREST has been used for open data before, but the way it's setup on Supabase cloud doesn't allow it because its proxy mandates the presence of an API key. Self hosting and removing that requirement could be one way to go.

**Reply by PierreMesure** [reply]
Thanks for your answer @steve-chavez! I understood a bit later that this was exactly what db-anon-role was about with PostgREST and that there isn't any API key needed there unlike on Supabase.

On Supabase, would it be possible to achieve it with the self-hosted project without forking and changing the way requests are handled without API key headers?

To be fair, I was mostly curious if this was possible out of the box but we are going to expose Supabase through a reverse proxy anyway (Caddy, most likely).

So we just need to [add the API key headers](https://caddyserver.com/docs/caddyfile/directives/header) artificially there.

Something like `header_up apiKey xxx...` in the Caddyfile should work I guess, I haven't tried. 🙂

**Reply by steve-chavez** [reply]
> On Supabase, would it be possible to achieve it with the self-hosted project without forking and changing the way requests are handled without API key headers?

For now you'd have to fork. I haven't seen much demand for open data projects on Supabase, but maybe you could open a discussion and see if there are others that'd be interested.
